src/test/run-pass/issues/issue-57084-async-await.rs

```diff
@@ -0,0 +1,27 @@
+// This issue reproduces an ICE on compile (E.g. fails on 2018-12-19 nightly).
+// run-pass
+// edition:2018
+#![feature(async_await,futures_api,await_macro,generators)]
+
+pub struct Foo;
+
+impl Foo {
+    async fn with<'a, F, R>(&'a self, f: F) -> R
```

Comment:
If you could make a reproducer that didn't depend on `async_await`/`futures_api`/`await_macro` and only on generators that would be appreciated since otherwise i'm going to have to go edit this soon when I change `await!` so that it is no longer a simple macro_rules! macro.

Reply:
(aside from that-- thanks for putting this change together! :) )

Reply:
Thanks, good idea - then it definitely belongs in the tests/run-pass/generators dir. 

After expanding the macro I could minimise it a little further so that pinning wasn't brought in, but the same DebruijnIndex error was raised.

Reply:
Sorry, to clarify: I didn't mean to just transition to `yield`, but remove `async` entirely here-- this is just an issue with generators.

Reply:
Hopfully this is what you were after?

Reply:
Looks great, thanks so much! :)